File: project-euler.java/src/main/java/euler/solutions/_12.java
package euler.solutions;

import euler.common.BaseSolution;

import static euler.common.NumberUtils.getDivisors;

public class _12 extends BaseSolution {

    @Override
    public long solve() {
        int number = 0;
        int term   = 0;
        while (true) {
            number++;
            term += number;
            if (getDivisors(term).length > 500) {
                return term;
            }
        }
    }
}

File: project-euler.go/solutions/_12/12.go
package _12

import "project-euler/common/numbers"

func Solve() int64 {
	number := 0
	term   := 0
	for {
		number++
		term += number
		divisors := numbers.GetDivisors(term)
		if len(divisors) > 500 {
			return int64(term)
		}
	}
}
